docs/pages/access-controls/reference.mdx

```diff
@@ -418,14 +320,253 @@ spec:
+- `kubernetes_labels_expression`
+- `db_labels_expression`
+- `db_service_labels_expression`
+- `windows_desktop_labels_expression`
+- `group_labels_expression`
+
+#### Variables
+
+You can use the following template variables in label expression fields:
+
+|Variable|Type|Description|
+|---|---|---|
+|`{{user.spec.traits}}`|`map[string][]string`|All of the user's traits. Use `{{user.spec.traits<key>}}` to get a list of trait values for the specified key.|
+|`{{labels}}`|`map[string]string`|All of the user's labels. Use `{{labels.<key>}}` to get the value for the specified key.|
+
+#### Functions
+
+You can use the following functions in label expression fields:
+
+|Function|Description|
+|---|---|
+|`{{contains(list,item)}}`|Returns `true` if the list of strings in `list` contains the specified `item` and `false` otherwise.|
+|`{{regexp.match(list,regexp)}}`|Returns `true` if the regular expression in `regexp` matches any of the strings in list of strings `list`. Returns `false` otherwise.|
+|`{{regexp.replace(list,regexp,replacement)}}`|Checks the list of strings `list`, returning a new list where each item that matches `regexp` is replaced with `replacement`. You can use `$n` in the replacement string to indicate a submatch group, e.g., `$1` to indicate the first submatch.|
+|`{{strings.upper(list)}}`|Returns the new list of strings that results from uppercasing each item in `list`, a list of strings.|
+|`{{strings.lower(list)}}`|Returns the new list of strings that results from lowercasing each item in `list`, a list of strings.|
+
+### Predicates for access to infrastructure resources
+
+When a user attempts to authenticate to an infrastructure resource
+proxied by Teleport, such as a database or Kubernetes cluster, Teleport determines from the user's roles:
+- Which infrastructure resources the user is authorized to authenticate to. 
+- Which principals the user can assume when they authenticate (for example,
+  Linux server logins and Kubernetes groups).
+
+#### Fields
+
+The following role fields control a user's access to infrastructure resources.
+All of these are fields within the `allow` and `deny` sections of a Teleport
+role resource.
+
+Labels for resources enrolled with Teleport:
+- `app_labels`
+- `cluster_labels`
+- `db_labels`
+- `db_service_labels`
+- `kubernetes_labels`
+- `node_labels`
+- `windows_desktop_labels`
+
+Principals a user can assume on third-party systems:
+- `aws_role_arns`
+- `azure_identities`
+- `db_names`
+- `db_roles`
+- `db_users`
+- `desktop_groups`
+- `gcp_service_accounts`
+- `host_groups`
+- `host_sudoers`
+- `kubernetes_groups`
+- `kubernetes_users`
+- `logins`
+- `windows_desktop_logins`
+
+Teleport principals that user can impersonate:
+- `impersonate.rules`
+- `impersonate.users`
+
+#### Functions
+
+You can use the following functions in role fields that govern access to
+principals on external systems. Both functions can take a template variable
+named above as an argument:
```

Comment:
```suggestion
as an argument:
```

Reply:
Removed this sentence, since the list of variables is below the list of functions.